### DISCUSSION TITLE
Subtitle size with css

### DISCUSSION THREAD

**Discussion by Dacogetul** [openplayerjs/openplayerjs#417] [General]
I don't know what else to do to be able to change the font sizes of the subtitles depending on the screen size. I tried to add a style.css file in the head and in the file I tried the @media-screen option for example:

`@media screen and (max-width: 360px) {
.op-captions {
 font-size: 12px !important;
 }
}

@media screen (min-width: 361px) and (max-width: 800px) {
.op-captions {
 font-size: 16px !important;
 }
}

@media screen (min-width: 801px) and (max-width: 1080px) {
.op-captions {
 font-size: 24px !important;
 }
}
@media screen (min-width: 1081px) and (max-width: 1800px) {
.op-captions {
 font-size: 26px !important;
 }
}
@media screen and (min-width: 1801px) {
.op-captions {
 font-size: 28px !important;
 }
}
`
and don't working... Can somebody show me how can do that..?

**Comment by rafa8626** [comment]
@Dacogetul is this for full screen mode? Do you have URL to check this?

**Comment by Dacogetul** [comment]
[Here](https://www.workpage.ro/test.php) is the link...
More than that, with new(last) openplayer.js or openplayer.min.js cannot change subtitle language....most stop for many time and change language too from button/s for i did it.
If cannot access the link..please tell me your ip for unlock from firewall....Thank you.. Or i can sent page php script with all css and js file...but in that case most change image, subtitle and movie link .... or try to open from phone without router wireless connection. (Prevent some ip if is blocked).
Thank you very much.

**Comment by Dacogetul** [comment]
I want to tell you ( because the openplayerjs have font size =0 ) I set font size 16px as default for the entire player . Without that your player settings show just arrow like ">". If i change this font(16px) can change all app font ... settings subtitle etc. not just one of this. Can show me a css file to setup font size separately for player or /and subtitle..???
Without posibility to change font size from settings i need to make this responsive.

**Reply by Dacogetul** [reply]
I managed to change the size of the subtitles separately from those of the player for different sizes of the browser, but I can't keep the proportion of the layer in the browser when I switch to fullscreen mode and I can't bring the layer with the subtitles above the bottom of the player either in openplayer or in fullscreen of another device. i make openplayermain.css with this code:
`.op-player {
    background: #000;
    font-family: sans-serif;
    font-size: 18px;
    line-height: 1.15;
    min-height: 30px;
    position: relative;

}
@media screen and (max-width: 480px) {
.op-captions {
 font-size: 70% !important; 
 }
}

@media screen and (min-width: 481px) and (max-width: 1080px) {
.op-captions {
 font-size: 100% !important;
 }
}


@media screen and (min-width: 1081px) and (max-width: 1800px) {
.op-captions {
 font-size: 150% !important;
 }
}

@media screen and (min-width: 1801px) {
.op-captions {
 font-size: 180% !important; 
}
}`

**Comment by rafa8626** [comment]
@Dacogetul for full screen more you are using the native cue element, not the one generated by the player. Check in the CSS sheet rules associated with the pseudoselector cue

**Comment by Dacogetul** [comment]
I don't how to do that im notprogrammer but just a novice...sorry...
is this ? : ::cue{color:#ccc}video:-webkit-media-text-track-display{top:-15%}. 
Can help me?

**Comment by rafa8626** [comment]
Yes those will help you to set the font size for the native captions

**Comment by Dacogetul** [comment]
If I understood correctly, this means to directly modify the settings in the openplayer.min.css file? I did not find anything in the cue: which contains a caption or subtitle so that I can set the font and the distance of its layer above the settings bar or the bottom edge of the screen

**Reply by rafa8626** [reply]
You can do that or in a separate styles tag where you can set the new CSS rules to override the result

**Comment by Dacogetul** [comment]
I think you're joking with me... :) I told you that I don't know how to program anymore... I thought you could help me with a small css code in which I can change the values until I reach the expected result. What I achieved with the css above I found on the internet and the value with .op-caption I took from css. I tried to change the bottom value for each op.caption but I didn't get anything.

**Reply by rafa8626** [reply]
Ok I’ll post something later to see if that helps you

**Reply by Dacogetul** [reply]
Thank you very much !!!!

**Reply by rafa8626** [reply]
@Dacogetul see if this helps
```html
<!DOCTYPE html>
<html>
    <head>
        <title>Test</title>
        <link href="https://cdn.jsdelivr.net/npm/openplayerjs@latest/dist/openplayer.min.css" rel="stylesheet" />
        <style>
            .op-player {
                background: #000;
                font-family: sans-serif;
                font-size: 18px;
                line-height: 1.15;
                min-height: 30px;
                position: relative;
            }
            .op-captions {
                font-size: 2.25vw; /* This unit will make the font responsive by using the viewport as reference */
            }
            ::cue {
                font-size: 45px;
            }
        </style>
    </head>
    <body>
        <video class="op-player__media" id="video" controls playsinline preload="none">
            <source src="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/TearsOfSteel.mp4" type="video/mp4">
            <track default srclang="it" label="Italian" src="./subtitles.vtt" kind="subtitles" />
        </video>
        <script src="https://cdn.jsdelivr.net/npm/openplayerjs@latest/dist/openplayer.min.js"></script>
        <script>
            const player = new OpenPlayerJS('video');
            player.init();
        </script>
    </body>
</html>
```

**Reply by Dacogetul** [reply]
Thank you!!! All work perfect with 1 exception: caption layer is right in the bottom in fullscreen after control hide. What can i do to muve up? 
![1](https://github.com/openplayerjs/openplayerjs/assets/18350449/745c084a-8d7e-41f5-a3b7-33e8181d1729)

**Reply by rafa8626** [reply]
@Dacogetul just increase the bottom property of the captions element
```css
.op-captions {
     bottom: 15%;  // or whatever you want to move it top
}
```

**Reply by Dacogetul** [reply]
This solution does not change the distance in fullscreen mode.

**Reply by Dacogetul** [reply]
Whit any value

**Reply by rafa8626** [reply]
Are you talking about the iOS fullscree mode?

**Reply by Dacogetul** [reply]
Is abou fullscreen in any device.
I add in css folder a file with openplayermain.css with this content:
` .op-player {
                background: #000;
                font-family: sans-serif;
                font-size: 18px;
                line-height: 1.15;
                min-height: 30px;
                position: relative;
            }
.op-captions {
                font-size: 2.25vw; /* This unit will make the font responsive by using the viewport as reference */
                bottom: 50%;  /* or whatever you want to move it top*/
            }
 ::cue {
                font-size: 45px;
                
            }'
with 50% bottom setting i have this when show the control bar:
![Screenshot 2024-04-29 072754](https://github.com/openplayerjs/openplayerjs/assets/18350449/631905d0-d0f3-4c35-b2a0-faf09138de10)

and this after control bar hide:
![Screenshot 2024-04-29 072809](https://github.com/openplayerjs/openplayerjs/assets/18350449/e1368de3-0c32-4921-b6ed-cd2093822e14)
This is for all device. MORE THAN: cannot change the LANGUAGE. I talk about default movie video.mp4 file and subtitles .vtt files. 
This file are default on the player not loaded with my (son) js script. :)  
For testing is same [link](https://workpage.ro/test.php)

**Reply by rafa8626** [reply]
@Dacogetul for the first issue, add this style after the other ones I provided you:

```css

.op-controls--hidden .op-captions {
    bottom: 0; // adjust it as you like
} 
```
This won’t affect the captions in iOS fullscreen and there’s not much to do about it, unless JS code is used for that.

For the second issue, can you post the link again, since I tried to click it and nothing happens?

**Reply by Dacogetul** [reply]
Thank you...can try now....my windows os make update and server do not autorestar after that...work now...please try again.... I make your settings and work in fullscreen mod to change/keep position.....Is it necessary to keep the css settings from .op-captions or are those from .op-controls--hidden .op-captions sufficient?  Now my css are this:
`.op-player {
    background: #000;
    font-family: sans-serif;
    font-size: 18px;
    line-height: 1.15;
    min-height: 30px;
    position: relative;

}
.op-controls--hidden .op-captions {
    bottom: 25%; // adjust it as you like
    font-size: 2.25vw; /* This unit will make the font responsive by using the viewport as reference */
    bottom: 3%;  /* or whatever you want to move it top*/
} 
.op-captions {
                font-size: 2.25vw; /* This unit will make the font responsive by using the viewport as reference */
                bottom: 3%;  /* or whatever you want to move it top*/
            }
 ::cue {
                font-size: 45px;
                
            }`

**Reply by Dacogetul** [reply]
PS : just default player have english subtitle.vtt  like engleza.vtt !!!!

**Reply by rafa8626** [reply]
You can leave both styles; that will ensure you are covered for all scenarios

**Reply by Dacogetul** [reply]
Why cannot change subtitle language with new version?

**Comment by rafa8626** [comment]
I can change it if I have a different content in the English file. I assume it must be an issue with the VTT file itself.

**Reply by Dacogetul** [reply]
I don't think it's from the subtitles... except that the text has been changed. I made 2 videos, both without my javascript code (to load and play the videos from the page). In the first movie you can see how the player doesn't even block (interrupt) the subtitle or change it, and in the second movie you can see how I changed some characters from the working file that I replaced in the player and it still does not work.
[Video1](https://drive.google.com/file/d/1nFqsjvOfTVxKSM2fpQqMtIcdwJKHwx1L/view?usp=drive_link)
[Video2](https://drive.google.com/file/d/1qzIwaPkH27REE6PvjTfIQlzy1uQPdX0b/view?usp=drive_link)

**Reply by Dacogetul** [reply]
For you I have made two identical pages, the first called test in which the new version of openplayer is used and takes the data from the dist folder and the second called test 1 in which the previous version of openplayer is used and which works perfectly less on IOS. ...
First version [test](https://workpage.ro/test.php)
Second version [test1](https://workpage.ro/test1.php)

**Reply by rafa8626** [reply]
I’ll take a look at this next week; I changed the approach to use native cues rather than trying to fetch them. I still find it odd that you have a problem with your captions but the ones I’ve been using don’t present that problem

**Reply by rafa8626** [reply]
@Dacogetul I cannot access either the videos or the website. I tested it again, switching languages, and I was able to get the switching properly. The only thing I'm going to add is to remove the old captions when switching languages, so we prep the container to receive new captions

**Reply by Dacogetul** [reply]
Okay....then I put the 2 pages with js and css and related video in the archive : [HERE](https://mega.nz/file/tAwGWBKD#wD786Fzwcv4NvBcNf8QvTLqtdHHbyGuCFSSKUvcVhsk)

**Reply by rafa8626** [reply]
@Dacogetul I'm moving this conversation to the issue area. I'm closing this and opening a bug
